test/source/tests/unit-node.ts

```diff
@@ -776,19 +775,40 @@ jLwe8W9IMt765T5x5oux9MmPDXF05xHfm4qfH/BMO3a802x5u2gJjJjuknrFdgXY
       t.pass();
     });
 
-    ava.default('[unit][MsgUtil.decryptMessage] extracts Primary User ID from key', async t => {
+    ava.default('[unit][MsgUtil.decryptMessage] finds correct key to verify signature', async t => {
       const data = await GoogleData.withInitializedData('ci.tests.gmail@flowcrypt.test');
       const msg: GmailMsg = data.getMessage('1766644f13510f58')!;
       const enc = Buf.fromBase64Str(msg!.raw!).toUtfStr()
         .match(/\-\-\-\-\-BEGIN PGP SIGNED MESSAGE\-\-\-\-\-.*\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-/s)![0];
       const encryptedData = Buf.fromUtfStr(enc);
-      const pubkey = await KeyUtil.parse(testConstants.pubkey2864E326A5BE488A);
-      await ContactStore.update(undefined, 'president@forged.com', { name: 'President', pubkey });
-      const decrypted = await MsgUtil.decryptMessage({ kisWithPp: [], encryptedData });
-      expect(decrypted.success).to.equal(true);
-      const verifyRes = (decrypted as DecryptSuccess).signature!;
-      expect(verifyRes.match).to.be.true;
-      expect(verifyRes.signer?.primaryUserId).to.equal('A50 Sam <sams50sams50sept@gmail.com>');
+      // actual key the message was signed with
+      const pubkey = testConstants.pubkey2864E326A5BE488A;
+      // better key
+      const betterKey = "-----BEGIN PGP PUBLIC KEY BLOCK-----\r\nVersion: FlowCrypt Email Encryption [BUILD_REPLACEABLE_VERSION]\r\nComment: Seamlessly send and receive encrypted email\r\n\r\nxjMEYZeW2RYJKwYBBAHaRw8BAQdAT5QfLVP3y1yukk3MM/oiuXLNe1f9az5M\r\nBnOlKdF0nKnNJVNvbWVib2R5IDxTYW1zNTBzYW1zNTBzZXB0QEdtYWlsLkNv\r\nbT7CjwQQFgoAIAUCYZeW2QYLCQcIAwIEFQgKAgQWAgEAAhkBAhsDAh4BACEJ\r\nEMrSTYqLk6SUFiEEBP90ux3d6kDwDdzvytJNiouTpJS27QEA7pFlkLfD0KFQ\r\nsH/dwb/NPzn5zCi2L9gjPAC3d8gv1fwA/0FjAy/vKct4D7QH8KwtEGQns5+D\r\nP1WxDr4YI2hp5TkAzjgEYZeW2RIKKwYBBAGXVQEFAQEHQKNLY/bXrhJMWA2+\r\nWTjk3I7KhawyZfLomJ4hovqr7UtOAwEIB8J4BBgWCAAJBQJhl5bZAhsMACEJ\r\nEMrSTYqLk6SUFiEEBP90ux3d6kDwDdzvytJNiouTpJQnpgD/c1CzfS3YzJUx\r\nnFMrhjiE0WVgqOV/3CkfI4m4RA30QUIA/ju8r4AD2h6lu3Mx/6I6PzIRZQty\r\nLvTkcu4UKodZa4kK\r\n=7C4A\r\n-----END PGP PUBLIC KEY BLOCK-----\r\n";
```

Comment:
What is the difference between the keys, how is the other one better? Would be good to explain in comments.

Reply:
The newest (the greater date of creation) was supposed to be "better", but currently this field isn't used in sorting. Should we take it into account?

Reply:
Let's name them `olderVersionOfPubkey` and `newerVersionOfPubkey` or similar. Better is unclear

Reply:
Done. Although they are not really versions -- they are compoletely independent. Is `newlyCreatedPubkey` good enough?

Reply:
> The newer (the greater date of creation) was supposed to be "better", but currently this field isn't used in sorting. Should we take it into account?

Newer key is only better if both have the exact same primary fingerprint, and therefore the newer is an updated version of the old. There is nothing better about a newly created different key. It's just another key.\

Do we have anywhere in our codebase a preference for newer keys that have a different primary fingerprint? If yes, that has to be removed (sounds like a security problem).

For this test, let's call the second key simply `wrongKey`. Since it's a key unrelated to the signature that was made. Whether it's newer or not should not matter.

Reply:
> Do we have anywhere in our codebase a preference for newer keys that have a different primary fingerprint? If yes, that has to investigated and discussed.

No, it turns out we don't have such preference.
We sort by expiration date. Both keys in this test are non-expirable, so they are considered equivalent

Reply:
Got it. Let's use `wrongKey` as a name for the second one.

Reply:
Done